packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -426,6 +432,41 @@ export const useGridColumnResize = (
     apiRef.current.forceUpdate();
   }, [apiRef]);
 
+  /**
+   * API METHODS
+   */
+
+  const autosizeColumns = React.useCallback<GridColumnResizeApi['autosizeColumns']>(
```

Comment:
I'm not sure which row sample to use. Currently I only use the rows rendered, but this sample might be too small to be representative. Is there a number we'd rather use as the default? 50? 100?

I can also include an option to make it configurable.

Reply:
I've set it to 40 right now, but the API call feels quite slow. But if we have less rows, we have less representative data.

Reply:
Alright, I'll leave that out for now.

For the record, if we end up doing this we should add hidden rows to the DOM if we need a bigger sample. Trying to add rows around the rendered region can mess up things because autosizing is an async operations and other stuff can change the region during that period, so it's easier to avoid concurrency issues that way.

Reply:
Targeting rendered rows should be enough for most use cases.
For the benchmark:
- Google Sheets: measures all cells (tested with 1k rows).
  > Note: They use canvas rendering and cell content is pretty much limited to text, so maybe this is why they can do this effectively
- AG Grid: [measures rendered rows only](https://www.ag-grid.com/javascript-data-grid/column-sizing/#auto-size-columns)
- SyncFusion: measures rendered rows only: https://stackblitz.com/edit/react-8nksr3?file=index.js

Reply:
> Note: They use canvas rendering and cell content is pretty much limited to text, so maybe this is why they can do this effectively

Yeah I figured we could also use `getComputedStyles` + canvas to figure out text width without the DOM, but we need the DOM anyway for non-text cells so it felt like it could add more complexity. I don't think it's unviable though, it would be easy to use that strategy for raw text column types and fall back to the DOM for the others.